### PR TITLE
feat: enableSwaps option

### PR DESCRIPTION
**Pull request by enesozturk**
# Description

Introduces `enableSwaps` flag which allow users to enable/disable swap feature. Also we are adding swap feature to the default account view so even with regular wallets, users will be able to use swaps.

By default, swaps are disabled on regular flow, enabled on email flow.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-752, Closes, APKT-618
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections

### CODE REVIEW

File: packages/core/src/controllers/SwapController.ts [hunk 1]
```diff
@@ -655,16 +660,20 @@ export const SwapController = {
...
+    if (isAuthConnector) {
+      RouterController.pushTransactionStack({
+        view: null,
+        goBack: true,
+        onSuccess() {
+          SnackController.showLoading('Approving transaction...')
+        }
+      })
+    } else {
+      SnackController.showLoading('Approve spending increase')
```
Comment (tomiir) [inline packages/core/src/controllers/SwapController.ts:675]: Do we need any loading indicator on the screen as well?

File: packages/core/src/controllers/SwapController.ts [hunk 2]
```diff
@@ -702,14 +714,18 @@ export const SwapController = {
...
+      RouterController.pushTransactionStack({
+        view: 'Account',
+        goBack: false,
+        onSuccess() {
+          SnackController.showLoading(snackbarPendingMessage)
+          SwapController.resetState()
+        }
+      })
+    } else {
+      SnackController.showLoading(snackbarPendingMessage)
```
Comment (tomiir) [inline packages/core/src/controllers/SwapController.ts:727]: same

File: packages/core/src/controllers/SwapController.ts [hunk 3]
```diff
@@ -739,6 +755,9 @@ export const SwapController = {
         }
       })
       SwapController.resetState()
+      if (!isAuthConnector) {
+        RouterController.replace('Account')
+      }
```
Comment (tomiir) [inline packages/core/src/controllers/SwapController.ts:760]: What happens if it's the auth connector?

Comment (enesozturk) [reply packages/core/src/controllers/SwapController.ts:760]: If it's auth connector, we are handing the success and error states on top with `RouterController.pushTransactionStack`, there is no need to call something here for auth connectors bc the way they are working is different.

File: packages/core/src/controllers/SwapController.ts (hunk 1 above)
Comment (enesozturk) [reply packages/core/src/controllers/SwapController.ts]: Yes, this is just a feedback to the users that let them know they need to approve the TX, toast will be visible for few seconds.

File: packages/core/src/controllers/SwapController.ts (hunk 2 above)
Comment (enesozturk) [reply packages/core/src/controllers/SwapController.ts]: Yes, this is just a feedback to the users that let them know they need to approve the TX, toast will be visible for few seconds.

**Review by tomiir — APPROVED** [review]

**Pull request merged**